x-pack/filebeat/input/http_endpoint/validate.go

```diff
@@ -5,43 +5,59 @@
 package http_endpoint
 
 import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha1"
+	"encoding/hex"
 	"errors"
 	"fmt"
+	"io/ioutil"
 	"net/http"
 )
 
 type validator interface {
-	// ValidateHeader checks the HTTP headers for compliance. The body must not
-	// be touched.
-	ValidateHeader(*http.Request) (int, error)
+	Validate(*http.Request) (int, error)
 }
 
-type apiValidator struct {
+type authValidator struct {
 	basicAuth          bool
 	username, password string
-	method             string
-	contentType        string
 	secretHeader       string
 	secretValue        string
 }
 
+type headerValidator struct {
+	method      string
+	contentType string
+}
```

Comment:
all validators are essentially header validators. Consider to rename this one into method validator or  split this one into method and content-type validators.